include/zephyr/fs/fs.h

```diff
@@ -162,6 +162,8 @@ struct fs_statvfs {
 #define FS_O_APPEND     0x20
 /** Bitmask for open/create flags */
 #define FS_O_FLAGS_MASK 0x30
+/** Truncate the file while opening */
+#define FS_O_TRUNC      0x40
```

Comment:
This should be after `FS_O_APPEND`, you also need to fix the `FS_O_FLAGS_MASK` that does not cover the flag.